src/sparseml/optim/manager.py

```diff
@@ -622,6 +622,54 @@ def phase(self, epoch: float) -> Optional[str]:
             else:
                 return "quantized_pruned"
 
+    def get_start_end_epochs(self) -> Dict[str, Tuple[float, float]]:
+        """
+        Return an OrderedDict mapping each stage to its min and max epoch. If not a
+        staged manager, map 'all' to the the min and max epochs
+        """
+        if isinstance(self.modifiers, List):
+            return OrderedDict({"all": (self.min_epochs, self.max_epochs)})
+        else:
+            stage_max_min = OrderedDict()
+            for stage, mod_list in self.modifiers.items():
+                epoch_floors = [
+                    math.floor(mod.start_epoch)
+                    for mod in mod_list
+                    if mod.start_epoch > -1
+                ] + [
+                    math.floor(mod.end_epoch) for mod in mod_list if mod.end_epoch > -1
+                ]
+                epoch_min = min(epoch_floors) if epoch_floors else -1
+
+                epoch_ceils = [
+                    math.ceil(mod.start_epoch)
+                    for mod in mod_list
+                    if mod.start_epoch > -1
+                ] + [math.ceil(mod.end_epoch) for mod in mod_list if mod.end_epoch > -1]
+                epoch_max = max(epoch_ceils) if epoch_ceils else -1
+
+                stage_max_min[stage] = (epoch_min, epoch_max)
+
+            # Post-process to replace -1's with their real values
+            epochs_list = list(stage_max_min.values())
+            for i, stage, epochs in enumerate(stage_max_min.items()):
+                if epochs[0] == -1:
+                    stage_max_min[stage][0] = epochs_list[i - 1][1] if i > 0 else 0
+                if epochs[1] == -1 and i < len(epochs_list) - 1:
```

Comment:
By this logic, if there is no explicit end epoch on the last stage, we're keeping it at -1. This is the only -1 we would potentially keep. Comment if you think this should be handled differently